### PR TITLE
Adds approx. a billion new gun related sounds

### PR DESCRIPTION
**Pull request by KingOfThePing**
This PR aims to make le epic gun handling even better.

Sound variety is nice to have and guns have nice sounds, when you handle them. This PR aims to replace some currently weak sounds (example: reloading), but mostly to add a bigger variety of sounds available.

W I P so I don't forget to work on it.

All sounds are either CC0, by-nc, or self recorded. 95% is CC0 though.

- [x] reload sounds for pistols
- [x] reload sounds for revolvers
- [x] reload sounds for shotguns
- [x] reload sounds for rifles
- [x] empty reload sounds (also known as "dry reloads") for shotguns >>> possible but no thanks
- [x] casing sounds
- [x] shotgun shell casing sounds
- [x] empty (dry firing) sounds
- [x] all of the above for some specialized ballistic weapons (e.g. grenade launchers, improvised guns)

### CODE REVIEW

**Comment by Zelmana** [comment]
Discussed this on discord for self recorded sounds, feel free to reach out.

**Comment by KingOfThePing** [comment]
!review

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -434,8 +436,11 @@
 	if(user)
 		to_chat(user, SPAN_DANGER("*click*"))
 	else
-		src.visible_message("*click click*")
-	playsound(loc, 'sound/weapons/empty.ogg', 100, 1)
+		src.visible_message("*click*")
+	playsound(loc, empty_sound, 100, 1)
+
+
+
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/gun.dm:443]: Unnecessary whitespace

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 2]
```diff
@@ -276,3 +277,12 @@
...
 	return FALSE
+
+
+/*
+/obj/item/gun/proc/handle_click_empty(mob/user)
+	if(user)
+		to_chat(user, SPAN_DANGER("*click*"))
+	else
+		src.visible_message("*click click*")
+	playsound(loc, 'sound/weapons/empty.ogg', 100, 1) */
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/guns/projectile/revolver.dm:288]: why is this commented out? If it's not functional, delete it. Also - pointless whitespace above.

File: code/modules/projectiles/gun.dm [hunk 3]
```diff
@@ -83,6 +83,8 @@
 	var/displays_maptext = FALSE
 	var/can_ammo_display = TRUE
 	var/obj/item/ammo_display
+	var/empty_sound = /decl/sound_category/out_of_ammo
+	var/casing_drop_sound = /decl/sound_category/casing_drop_sound
```
Comment (alsoandanswer) [inline code/modules/projectiles/gun.dm:87]: Redundant var is redundant - why not just use the casing drop sound...that's already on the casing?

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]
Most of the sounds look fine.

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 4]
```diff
@@ -84,7 +85,7 @@
 
 	if(chambered)//We have a shell in the chamber
 		chambered.forceMove(get_turf(src)) //Eject casing
-		playsound(src.loc, chambered.drop_sound, DROP_SOUND_VOLUME, FALSE, required_asfx_toggles = ASFX_DROPSOUND)
+		playsound(src.loc, casing_drop_sound, DROP_SOUND_VOLUME, FALSE, required_asfx_toggles = ASFX_DROPSOUND)
```
Comment (alsoandanswer) [inline code/modules/projectiles/guns/projectile/shotgun.dm:88]: Same as above

File: code/modules/projectiles/guns/projectile/revolver.dm (hunk 2 above)
Comment (KingOfThePing) [reply code/modules/projectiles/guns/projectile/revolver.dm:288]: Leftover from 2 weeks ago, just forgor

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 5]
```diff
@@ -4,6 +4,7 @@
 	var/can_sawoff = FALSE
 	var/sawnoff_workmsg
 	var/sawing_in_progress = FALSE
+	casing_drop_sound = /decl/sound_category/casing_drop_sound_shotgun
```
[the remark this answers is not in the document]
Comment (KingOfThePing) [reply code/modules/projectiles/guns/projectile/shotgun.dm:7]: Casings have different drop sounds now. Shotgun shells have their own drop sound. They do not sound like regular bullet casings

File: code/modules/projectiles/guns/projectile/shotgun.dm (hunk 4 above)
Comment (KingOfThePing) [reply code/modules/projectiles/guns/projectile/shotgun.dm:88]: When tested, shells from boxes played the sound fine - when pump-actioned out of a shotgun they played the incorrect (metallic) sound. This here fixed that, at least for me

File: code/modules/projectiles/gun.dm (hunk 3 above)
Comment (KingOfThePing) [reply code/modules/projectiles/gun.dm:87]: Oversight

Comment (KingOfThePing) [reply code/modules/projectiles/gun.dm]: Actually, no. This was no oversight. This is needed so shotgun shells can have their own drop sounds.

File: code/modules/projectiles/ammunition/bullets.dm [hunk 6]
```diff
@@ -336,7 +337,8 @@
 	w_class = ITEMSIZE_NORMAL
 	slot_flags = null
 	max_stack = 1
-	reload_sound = 'sound/weapons/reload_shell.ogg'
+	reload_sound = 'sound/weapons/reloads/shotgun_pump.ogg'
+	drop_sound = /decl/sound_category/generic_drop_sound
```
Comment (alsoandanswer) [inline code/modules/projectiles/ammunition/bullets.dm:341]: Shouldn't this be /decl/sound_category/casing_drop_sound_shotgun?

This might be the reason why the shotgun shells sound weird. Please remove  var/casing_drop_sound.

**Review by alsoandanswer — CHANGES_REQUESTED** [review]

**Review by Alberyk — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]
